### PR TITLE
chore: Snaps and Accounts selectors usage

### LINKED ISSUE #10312: chore: Abstract controller selectors in `accounts` and `snaps` related dirs
### What is this about?

This task is a prerequisite for abstracting controller selectors in preparation for optimizing selectors (memoization). This task focuses on files in `accounts` and `snaps` related dirs

### Scenario

_No response_

### Design

_No response_

### Technical Details

- [ ] app/components/Views/LedgerAccountInfo/index.tsx
  101,13:       state.engine.backgroundState.NetworkController.providerConfig,
  107,13:       state.engine.backgroundState.PreferencesController.frequentRpcList,

- [ ] app/components/Views/SDK/SDKSessionModal/SDKSessionModal.tsx
  97,27:     (state: any) => state.engine.backgroundState.PermissionController,

- [ ] app/components/Views/Settings/Contacts/ContactForm/index.js
  507,22:   addressBook: state.engine.backgroundState.AddressBookController.addressBook,

- [ ] app/components/Views/Settings/Contacts/index.js
  178,22:   addressBook: state.engine.backgroundState.AddressBookController.addressBook,

- [ ] app/components/Views/Snaps/SnapSettings/SnapSettings.tsx
  48,27:     (state: any) => state.engine.backgroundState.PermissionController,

- [ ] app/components/Views/Snaps/SnapsSettingsList/SnapsSettingsList.tsx
  27,27:     (state: any) => state.engine.backgroundState.SnapController.snaps,

### Threat Modeling Framework

_No response_

### Acceptance Criteria

- App behavior remains the same
- Spot check the flows where changes were made to confirm consistent behavior

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by tommasini**
## **Description**

Use existing selectors on accounts and snaps directories

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by Cal-L — APPROVED** [review]
LGTM

**Review by Jonathansoufer — APPROVED** [review]
LGTM!

File: app/components/Views/Settings/Contacts/ContactForm/index.js [hunk 1]
```diff
@@ -504,7 +505,7 @@ class ContactForm extends PureComponent {
 ContactForm.contextType = ThemeContext;
 
 const mapStateToProps = (state) => ({
-  addressBook: state.engine.backgroundState.AddressBookController.addressBook,
+  addressBook: selectAddressBook(state),
```
Comment (NicolasMassart) [inline app/components/Views/Settings/Contacts/ContactForm/index.js:508]: Q: why not using useSelector here? No need of memo?

Comment (tommasini) [reply app/components/Views/Settings/Contacts/ContactForm/index.js:508]: The useSelector is for functional components, and this is a class component!
Also, the memoization it happens at the selector level when we do `createSelector` imported with the `reselect` package

**Pull request merged**